include/ShortcutsWindow.hpp

```diff
@@ -0,0 +1,34 @@
+//================================================================================================
+/// @file ShortcutsWindow.hpp
+///
+/// @brief Defines a dialog where keyboard shortcuts could be selected
+/// @author Miklos Marton
+///
+/// @copyright The Open-Agriculture Developers
+//================================================================================================
+#pragma once
+
+#include "JuceHeader.h"
+
+class ShortcutsWindow : public juce::AlertWindow
+  , public juce::KeyListener
+{
+public:
+	ShortcutsWindow(int currentKeyCode, Component *associatedComponent = nullptr);
+
+	void resized() override;
+	bool keyPressed(const juce::KeyPress &key, juce::Component *originatingComponent) override;
+	int selectedKeyCode() const;
+
+private:
+	juce::TextButton selectKeyButton;
+	int selectedKeyCode_ = juce::KeyPress::escapeKey;
+
+	bool keySelectionMode = false;
+
+	void updateAlarmAckButtonLabel(const juce::KeyPress &key);
+
+	void setKeySelectionMode(bool isEnabled);
```

Comment:
Nit (non-blocking): We don't normally postfix our members with `_`. Except when inherited from Juce, types are usually `PascalCase` and methods are `snake_case` for our various C++ projects. Not super important but those conventions would be the most consistent with our other code.

Reply:
Yeah, I added the _ suffix because the name would have collide with the getter variable.
I looked around how the getters named and did not found any get prefixed getter (and I also prefer the get-less names) .
Anyway I played some chess with the variable naming by keeping in mind that at some day more shortcuts might need to be added, I think it will meet your expectations now.